src/react/useAtomValue.ts

```diff
@@ -113,6 +113,10 @@ export function useAtomValue<AtomType extends Atom<unknown>>(
 export function useAtomValue<Value>(atom: Atom<Value>, options?: Options) {
   const store = useStore(options)
 
+  if (atom === undefined || atom === null) {
```

Comment:
Let's make this DEV-only:

```suggestion
  if (import.meta.env?.MODE !== 'production' && (atom === undefined || atom === null)) {
```

Reply:
Maybe this is simpler:

```suggestion
  if (import.meta.env?.MODE !== 'production' && !atom) {
```

Reply:
In case we want stricter check, this is possible:

```suggestion
  if (import.meta.env?.MODE !== 'production' && typeof atom?.read !== 'function') {
```